ash/src/device.rs

```diff
@@ -1276,6 +1276,36 @@ pub trait DeviceV1_0 {
         }
     }
 
+    unsafe fn get_pipeline_cache_data(
+        &self,
+        pipeline_cache: vk::PipelineCache,
+    ) -> VkResult<Vec<u8>> {
+        let data_size: *mut usize = mem::uninitialized();
+        let data: *mut c_void = mem::uninitialized();
+        let err_code = self.fp_v1_0().get_pipeline_cache_data(
+            self.handle(),
+            pipeline_cache,
+            data_size,
+            data,
+        );
+
+        if err_code != vk::Result::SUCCESS {
+            return Err(err_code);
+        };
+
+        let err_code = self.fp_v1_0().get_pipeline_cache_data(
+            self.handle(),
+            pipeline_cache,
+            data_size,
+            data,
+        );
+
+        match err_code {
+            vk::Result::SUCCESS => Ok(Vec::from_raw_parts(data as _, *data_size, *data_size)),
```

Comment:
You have to allocate the buffer to `data` yourself. You can do this with `Vec::with_capacity` and `set_len`. And you need to  call `get_pipeline_cache_data` two times to get the size, and then to fill the buffer.